tests/threads_nested_attach_permanently.rs

```diff
@@ -47,7 +47,6 @@ pub fn nested_attaches_should_not_detach_permanent_thread() {
     assert_eq!(jvm().threads_attached(), 1);
 
     // Dropping JNIEnv obtained from a permanent attach has no effect on the thread status.
-    drop(env);
```

Comment:
Same here.

Reply:
It looks like a part of the test, but `drop` for `JNIEnv` doesn't do anything. We have several similar tests, but only one of them (`nested_attaches_should_not_detach_guarded_thread`) uses `AttachGuard`.

Reply:
Doesn't it make sense to leave it? It seems to verify that whatever `Drop` implementation JNIEnv has, it does not detach a permanently attached thread. At the moment it has no implementation, but if it ever changes, this test will verify that `attach_current_thread_permanently` contract is preserved: i.e., "permanently" means "until explicitly detached, and no through other means".

Reply:
The type cannot implement both `Copy` and `Clone`, so we cannot implement `Drop` without breaking changes (removing `Copy` implementation). We can silence this warning in order to preserve drop in the test, but I'm not sure if it makes sense. I guess extending the comment will be enough.

Reply:
I see, thanks for the explanation. Then I suggest to remove the redundant comments and assertions. @aleksuss could you do that please, or would you prefer to postpone these changes till a separate PR?